### PR TITLE
add spectacle docs generation

### PR DESCRIPTION
**Pull request by porteta**

### CODE REVIEW

File: api-docs/api.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+swagger: "2.0"
+info:
+  version: 2.4.0
+  title: Koverse API
+  description: >
+        Koverse Web API Specification
+
+
+
+host: koverse.swagger.io:8000
```
Comment (porteta) [inline api-docs/api.yaml]: Does this need to be changed?

File: api-docs/paths/dataSet.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  - $ref: "../api.yaml#/parameters/dataSetId"
+
+get:
+  summary: Get metadata
+  description: >
+        Get the name, description, record count, and other information
+        about the given data set. Does not return any records within the
+        data set.
+  operationId: findDataSetById
+  tags: ["DataSet"]
```
Comment (porteta) [inline api-docs/paths/dataSet.yaml]: I think we should use the same "DataSets" tag here so the operations are grouped together in the documentation

File: api-docs/schemas/dataSetMetadata.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+type: object
```
Comment (porteta) [inline api-docs/schemas/dataSetMetadata.yaml]: I think we should add example blocks for all of our schemas. These will make more sense in the docs than filling the values in as the types.

File: api-docs/api.yaml (hunk 1 above)
Comment (acordova) [reply api-docs/api.yaml]: Yes this should be changed to something that works but I don't know what

File: api-docs/paths/dataSet.yaml (hunk 2 above)
Comment (acordova) [reply api-docs/paths/dataSet.yaml]: These are technically two different paths - one for /api/datasets and the other for /api/datasets/{datasetId} .. there may be a method that conflicts as a result if we were to group them .. for instance just the GET .. but perhaps that's ok if they have different summaries?

File: api-docs/schemas/dataSetMetadata.yaml (hunk 3 above)
Comment (acordova) [reply api-docs/schemas/dataSetMetadata.yaml]: Ahh just found the docs on how to add examples. I'll do that for all the schemas

**Pull request merged**